src/test/java/org/wso2/lsp4intellij/utils/FileUtilsTest.java

```diff
@@ -0,0 +1,180 @@
+package org.wso2.lsp4intellij.utils;
+
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.TextEditor;
+import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.BinaryLightVirtualFile;
+import com.intellij.testFramework.LightVirtualFile;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import java.io.File;
+import java.net.URISyntaxException;
+
+@RunWith(PowerMockRunner.class)
+public class FileUtilsTest {
+
+  @PrepareForTest(FileEditorManager.class)
+  @Test
+  public void testEditorFromVirtualFile() {
+    VirtualFile file = PowerMockito.mock(VirtualFile.class);
+    Project project = PowerMockito.mock(Project.class);
+    
+    Editor editor = PowerMockito.mock(Editor.class);
+    TextEditor textEditor = PowerMockito.mock(TextEditor.class);
+    PowerMockito.when(textEditor.getEditor()).thenReturn(editor);
+
+    FileEditorManagerEx fileEditorManagerEx = PowerMockito.mock(FileEditorManagerEx.class);
+
+    PowerMockito.mockStatic(FileEditorManager.class);
+    PowerMockito.when(fileEditorManagerEx.getAllEditors(file))
+        .thenReturn(new FileEditor[]{textEditor})
+        .thenReturn(new FileEditor[0]);
+    PowerMockito.when(FileEditorManager.getInstance(project)).thenReturn(fileEditorManagerEx);
+
+    Assert.assertEquals(editor, FileUtils.editorFromVirtualFile(file, project));
+
+    Assert.assertNull(FileUtils.editorFromVirtualFile(file, project));
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testVirtualFileFromURI() {
+    LocalFileSystem localFileSystem = PowerMockito.mock(LocalFileSystem.class);
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenReturn(localFileSystem);
+    PowerMockito.when(LocalFileSystem.getInstance().findFileByIoFile(Mockito.any())).thenReturn(new BinaryLightVirtualFile());
```

Comment:
I think this line is longer than 80 chars

Reply:
I have cut some of the larger ones but i'm not cutting every one that is above 80 as we did say were not going to be so fussy on formatting and the repository owner does not seem to care about line length as many of his are well above 100